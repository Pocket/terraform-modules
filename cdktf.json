{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/src/example.js",
  "terraformProviders": [
    "pagerduty/pagerduty@~> 1.8.0",
    "archive@~> 1.3.0",
    "local@~> 2.1.0"
  ]
}
