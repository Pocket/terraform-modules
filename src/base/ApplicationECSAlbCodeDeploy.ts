import { CodedeployApp } from '@cdktf/provider-aws/lib/codedeploy-app';
import { CodedeployDeploymentGroup } from '@cdktf/provider-aws/lib/codedeploy-deployment-group';
import { CodestarnotificationsNotificationRule } from '@cdktf/provider-aws/lib/codestarnotifications-notification-rule';
import { DataAwsCallerIdentity } from '@cdktf/provider-aws/lib/data-aws-caller-identity';
import { DataAwsIamPolicyDocument } from '@cdktf/provider-aws/lib/data-aws-iam-policy-document';
import { DataAwsRegion } from '@cdktf/provider-aws/lib/data-aws-region';
import { IamRole } from '@cdktf/provider-aws/lib/iam-role';
import { IamRolePolicyAttachment } from '@cdktf/provider-aws/lib/iam-role-policy-attachment';
import { TerraformMetaArguments, TerraformResource } from 'cdktf';
import { Construct } from 'constructs';

export interface ApplicationECSAlbCodeDeployProps
  extends TerraformMetaArguments {
  prefix: string;
  clusterName: string;
  serviceName: string;
  listenerArn: string;
  snsNotificationTopicArn?: string;
  targetGroupNames: string[];
  tags?: { [key: string]: string };
  dependsOn?: TerraformResource[];
  successTerminationWaitTimeInMinutes?: number;
  notifications?: {
    notifyOnStarted?: boolean;
    notifyOnSucceeded?: boolean;
    notifyOnFailed?: boolean;
  };
}

interface CodeDeployResponse {
  codeDeployApp: CodedeployApp;
  ecsCodeDeployRole: IamRole;
}

/**
 * Represents a ecs Codeploy App that uses an ALB
 */
export class ApplicationECSAlbCodeDeploy extends Construct {
  private readonly config: ApplicationECSAlbCodeDeployProps;

  public readonly codeDeployApp: CodedeployApp;
  public readonly codeDeployDeploymentGroup: CodedeployDeploymentGroup;

  constructor(
    scope: Construct,
    name: string,
    config: ApplicationECSAlbCodeDeployProps,
  ) {
    super(scope, name);

    this.config = config;

    const { codeDeployApp, ecsCodeDeployRole } = this.setupCodeDeployApp();
    this.codeDeployApp = codeDeployApp;

    this.codeDeployDeploymentGroup = new CodedeployDeploymentGroup(
      this,
      `ecs_codedeploy_deployment_group`,
      {
        dependsOn: config.dependsOn,
        appName: codeDeployApp.name,
        deploymentConfigName: 'CodeDeployDefault.ECSAllAtOnce',
        deploymentGroupName: `${this.config.prefix}-ECS`,
        serviceRoleArn: ecsCodeDeployRole.arn,
        autoRollbackConfiguration: {
          enabled: true,
          events: ['DEPLOYMENT_FAILURE'],
        },
        blueGreenDeploymentConfig: {
          deploymentReadyOption: {
            actionOnTimeout: 'CONTINUE_DEPLOYMENT',
          },
          terminateBlueInstancesOnDeploymentSuccess: {
            action: 'TERMINATE',
            terminationWaitTimeInMinutes:
              (this.config.successTerminationWaitTimeInMinutes ??= 5),
          },
        },
        deploymentStyle: {
          deploymentOption: 'WITH_TRAFFIC_CONTROL',
          deploymentType: 'BLUE_GREEN',
        },
        ecsService: {
          clusterName: this.config.clusterName,
          serviceName: this.config.serviceName,
        },
        loadBalancerInfo: {
          targetGroupPairInfo: {
            prodTrafficRoute: { listenerArns: [this.config.listenerArn] },
            targetGroup: this.config.targetGroupNames.map((name) => {
              return { name };
            }),
          },
        },
        tags: this.config.tags,
        provider: this.config.provider,
      },
    );
  }

  /**
   * Set configuration for code deploy notifications
   * @param notifyOnStarted
   * @param notifyOnSucceeded
   * @param notifyOnFailed
   * @returns An array of EventTypeIds
   */

  private getEventTypeIds(
    notifyOnStarted = true,
    notifyOnSucceeded = true,
    notifyOnFailed = true,
  ): string[] {
    const eventTypeIds: string[] = [];

    if (notifyOnFailed) {
      eventTypeIds.push('codedeploy-application-deployment-failed');
    }

    if (notifyOnSucceeded) {
      eventTypeIds.push('codedeploy-application-deployment-succeeded');
    }

    if (notifyOnStarted) {
      eventTypeIds.push('codedeploy-application-deployment-started');
    }

    return eventTypeIds;
  }

  /**
   * Setup the codedeploy app, permissions, and notifications
   * @private
   */
  private setupCodeDeployApp(): CodeDeployResponse {
    const ecsCodeDeployRole = new IamRole(this, 'ecs_code_deploy_role', {
      name: `${this.config.prefix}-ECSCodeDeployRole`,
      assumeRolePolicy: new DataAwsIamPolicyDocument(
        this,
        `codedeploy_assume_role`,
        {
          statement: [
            {
              effect: 'Allow',
              actions: ['sts:AssumeRole'],
              principals: [
                {
                  identifiers: ['codedeploy.amazonaws.com'],
                  type: 'Service',
                },
              ],
            },
          ],
        },
      ).json,
      tags: this.config.tags,
      provider: this.config.provider,
    });

    new IamRolePolicyAttachment(this, 'ecs_codedeploy_role_attachment', {
      policyArn: 'arn:aws:iam::aws:policy/AWSCodeDeployRoleForECS',
      role: ecsCodeDeployRole.name,
      dependsOn: [ecsCodeDeployRole],
      provider: this.config.provider,
    });

    const codeDeployApp = new CodedeployApp(this, 'ecs_code_deploy', {
      computePlatform: 'ECS',
      name: `${this.config.prefix}-ECS`,
      tags: this.config.tags,
      provider: this.config.provider,
    });

    if (this.config.snsNotificationTopicArn) {
      const region = new DataAwsRegion(this, 'current_region', {
        provider: this.config.provider,
      });
      const account = new DataAwsCallerIdentity(this, 'current_account', {
        provider: this.config.provider,
      });
      new CodestarnotificationsNotificationRule(
        this,
        `ecs_codedeploy_notifications`,
        {
          detailType: 'BASIC',
          eventTypeIds: this.getEventTypeIds(
            this.config.notifications?.notifyOnStarted,
            this.config.notifications?.notifyOnSucceeded,
            this.config.notifications?.notifyOnFailed,
          ),
          name: codeDeployApp.name,
          resource: `arn:aws:codedeploy:${region.name}:${account.accountId}:application:${codeDeployApp.name}`,
          target: [
            {
              address: this.config.snsNotificationTopicArn,
            },
          ],
          tags: this.config.tags,
          provider: this.config.provider,
        },
      );
    }

    return { codeDeployApp, ecsCodeDeployRole };
  }
}
