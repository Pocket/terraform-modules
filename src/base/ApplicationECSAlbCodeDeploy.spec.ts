import { Testing } from 'cdktf';
import { ApplicationECSAlbCodeDeploy } from './ApplicationECSAlbCodeDeploy';

describe('ApplicationECSAlbCodeDeploy', () => {
  it('renders a CodeDeploy without a sns', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationECSAlbCodeDeploy(stack, 'testCodeDeploy', {
        prefix: 'Test-Dev',
        clusterName: 'cluster',
        serviceName: 'theService',
        listenerArn: 'listen-to-me',
        targetGroupNames: ['target-1', 'target-2'],
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders a CodeDeploy with a sns', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationECSAlbCodeDeploy(stack, 'testCodeDeploy', {
        prefix: 'Test-Dev',
        clusterName: 'cluster',
        serviceName: 'theService',
        listenerArn: 'listen-to-me',
        snsNotificationTopicArn: 'notify-me',
        targetGroupNames: ['target-1', 'target-2'],
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders a CodeDeploy with tags', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationECSAlbCodeDeploy(stack, 'testCodeDeploy', {
        prefix: 'Test-Dev',
        clusterName: 'cluster',
        serviceName: 'theService',
        listenerArn: 'listen-to-me',
        snsNotificationTopicArn: 'notify-me',
        targetGroupNames: ['target-1', 'target-2'],
        tags: {
          test: '1234',
          tag: 'me',
        },
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders a CodeDeploy with only failed deploy notifications', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationECSAlbCodeDeploy(stack, 'testCodeDeploy', {
        prefix: 'Test-Dev',
        clusterName: 'cluster',
        serviceName: 'theService',
        listenerArn: 'listen-to-me',
        snsNotificationTopicArn: 'notify-me',
        targetGroupNames: ['target-1', 'target-2'],
        tags: {
          test: '1234',
          tag: 'me',
        },
        notifications: {
          notifyOnFailed: true,
          notifyOnStarted: false,
          notifyOnSucceeded: false,
        },
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders a CodeDeploy with only succeeded deploy notifications', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationECSAlbCodeDeploy(stack, 'testCodeDeploy', {
        prefix: 'Test-Dev',
        clusterName: 'cluster',
        serviceName: 'theService',
        listenerArn: 'listen-to-me',
        snsNotificationTopicArn: 'notify-me',
        targetGroupNames: ['target-1', 'target-2'],
        tags: {
          test: '1234',
          tag: 'me',
        },
        notifications: {
          notifyOnFailed: false,
          notifyOnStarted: false,
          notifyOnSucceeded: true,
        },
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders a CodeDeploy with only started deploy notifications', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationECSAlbCodeDeploy(stack, 'testCodeDeploy', {
        prefix: 'Test-Dev',
        clusterName: 'cluster',
        serviceName: 'theService',
        listenerArn: 'listen-to-me',
        snsNotificationTopicArn: 'notify-me',
        targetGroupNames: ['target-1', 'target-2'],
        tags: {
          test: '1234',
          tag: 'me',
        },
        notifications: {
          notifyOnFailed: false,
          notifyOnStarted: true,
          notifyOnSucceeded: false,
        },
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders a CodeDeploy with no deploy notifications', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationECSAlbCodeDeploy(stack, 'testCodeDeploy', {
        prefix: 'Test-Dev',
        clusterName: 'cluster',
        serviceName: 'theService',
        listenerArn: 'listen-to-me',
        snsNotificationTopicArn: 'notify-me',
        targetGroupNames: ['target-1', 'target-2'],
        tags: {
          test: '1234',
          tag: 'me',
        },
        notifications: {
          notifyOnFailed: false,
          notifyOnStarted: false,
          notifyOnSucceeded: false,
        },
      });
    });
    expect(synthed).toMatchSnapshot();
  });
});
