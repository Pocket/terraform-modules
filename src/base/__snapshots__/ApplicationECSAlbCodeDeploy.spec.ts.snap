// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationECSAlbCodeDeploy renders a CodeDeploy with a sns 1`] = `
"{
  \\"data\\": {
    \\"aws_caller_identity\\": {
      \\"testCodeDeploy_current_account_0C0AC491\\": {
      }
    },
    \\"aws_iam_policy_document\\": {
      \\"testCodeDeploy_codedeploy_assume_role_E9FEDA7F\\": {
        \\"statement\\": [
          {
            \\"actions\\": [
              \\"sts:AssumeRole\\"
            ],
            \\"effect\\": \\"Allow\\",
            \\"principals\\": [
              {
                \\"identifiers\\": [
                  \\"codedeploy.amazonaws.com\\"
                ],
                \\"type\\": \\"Service\\"
              }
            ]
          }
        ]
      }
    },
    \\"aws_region\\": {
      \\"testCodeDeploy_current_region_B8533BBF\\": {
      }
    }
  },
  \\"resource\\": {
    \\"aws_codedeploy_app\\": {
      \\"testCodeDeploy_ecs_code_deploy_C4535318\\": {
        \\"compute_platform\\": \\"ECS\\",
        \\"name\\": \\"Test-Dev-ECS\\"
      }
    },
    \\"aws_codedeploy_deployment_group\\": {
      \\"testCodeDeploy_ecs_codedeploy_deployment_group_F092EC5E\\": {
        \\"app_name\\": \\"\${aws_codedeploy_app.testCodeDeploy_ecs_code_deploy_C4535318.name}\\",
        \\"auto_rollback_configuration\\": {
          \\"enabled\\": true,
          \\"events\\": [
            \\"DEPLOYMENT_FAILURE\\"
          ]
        },
        \\"blue_green_deployment_config\\": {
          \\"deployment_ready_option\\": {
            \\"action_on_timeout\\": \\"CONTINUE_DEPLOYMENT\\"
          },
          \\"terminate_blue_instances_on_deployment_success\\": {
            \\"action\\": \\"TERMINATE\\",
            \\"termination_wait_time_in_minutes\\": 5
          }
        },
        \\"deployment_config_name\\": \\"CodeDeployDefault.ECSAllAtOnce\\",
        \\"deployment_group_name\\": \\"Test-Dev-ECS\\",
        \\"deployment_style\\": {
          \\"deployment_option\\": \\"WITH_TRAFFIC_CONTROL\\",
          \\"deployment_type\\": \\"BLUE_GREEN\\"
        },
        \\"ecs_service\\": {
          \\"cluster_name\\": \\"cluster\\",
          \\"service_name\\": \\"theService\\"
        },
        \\"load_balancer_info\\": {
          \\"target_group_pair_info\\": {
            \\"prod_traffic_route\\": {
              \\"listener_arns\\": [
                \\"listen-to-me\\"
              ]
            },
            \\"target_group\\": [
              {
                \\"name\\": \\"target-1\\"
              },
              {
                \\"name\\": \\"target-2\\"
              }
            ]
          }
        },
        \\"service_role_arn\\": \\"\${aws_iam_role.testCodeDeploy_ecs_code_deploy_role_EF679F94.arn}\\"
      }
    },
    \\"aws_codestarnotifications_notification_rule\\": {
      \\"testCodeDeploy_ecs_codedeploy_notifications_7223DA1D\\": {
        \\"detail_type\\": \\"BASIC\\",
        \\"event_type_ids\\": [
          \\"codedeploy-application-deployment-failed\\",
          \\"codedeploy-application-deployment-succeeded\\",
          \\"codedeploy-application-deployment-started\\"
        ],
        \\"name\\": \\"\${aws_codedeploy_app.testCodeDeploy_ecs_code_deploy_C4535318.name}\\",
        \\"resource\\": \\"arn:aws:codedeploy:\${data.aws_region.testCodeDeploy_current_region_B8533BBF.name}:\${data.aws_caller_identity.testCodeDeploy_current_account_0C0AC491.account_id}:application:\${aws_codedeploy_app.testCodeDeploy_ecs_code_deploy_C4535318.name}\\",
        \\"target\\": [
          {
            \\"address\\": \\"notify-me\\"
          }
        ]
      }
    },
    \\"aws_iam_role\\": {
      \\"testCodeDeploy_ecs_code_deploy_role_EF679F94\\": {
        \\"assume_role_policy\\": \\"\${data.aws_iam_policy_document.testCodeDeploy_codedeploy_assume_role_E9FEDA7F.json}\\",
        \\"name\\": \\"Test-Dev-ECSCodeDeployRole\\"
      }
    },
    \\"aws_iam_role_policy_attachment\\": {
      \\"testCodeDeploy_ecs_codedeploy_role_attachment_0C2B6B97\\": {
        \\"depends_on\\": [
          \\"aws_iam_role.testCodeDeploy_ecs_code_deploy_role_EF679F94\\"
        ],
        \\"policy_arn\\": \\"arn:aws:iam::aws:policy/AWSCodeDeployRoleForECS\\",
        \\"role\\": \\"\${aws_iam_role.testCodeDeploy_ecs_code_deploy_role_EF679F94.name}\\"
      }
    }
  }
}"
`;

exports[`ApplicationECSAlbCodeDeploy renders a CodeDeploy with tags 1`] = `
"{
  \\"data\\": {
    \\"aws_caller_identity\\": {
      \\"testCodeDeploy_current_account_0C0AC491\\": {
      }
    },
    \\"aws_iam_policy_document\\": {
      \\"testCodeDeploy_codedeploy_assume_role_E9FEDA7F\\": {
        \\"statement\\": [
          {
            \\"actions\\": [
              \\"sts:AssumeRole\\"
            ],
            \\"effect\\": \\"Allow\\",
            \\"principals\\": [
              {
                \\"identifiers\\": [
                  \\"codedeploy.amazonaws.com\\"
                ],
                \\"type\\": \\"Service\\"
              }
            ]
          }
        ]
      }
    },
    \\"aws_region\\": {
      \\"testCodeDeploy_current_region_B8533BBF\\": {
      }
    }
  },
  \\"resource\\": {
    \\"aws_codedeploy_app\\": {
      \\"testCodeDeploy_ecs_code_deploy_C4535318\\": {
        \\"compute_platform\\": \\"ECS\\",
        \\"name\\": \\"Test-Dev-ECS\\"
      }
    },
    \\"aws_codedeploy_deployment_group\\": {
      \\"testCodeDeploy_ecs_codedeploy_deployment_group_F092EC5E\\": {
        \\"app_name\\": \\"\${aws_codedeploy_app.testCodeDeploy_ecs_code_deploy_C4535318.name}\\",
        \\"auto_rollback_configuration\\": {
          \\"enabled\\": true,
          \\"events\\": [
            \\"DEPLOYMENT_FAILURE\\"
          ]
        },
        \\"blue_green_deployment_config\\": {
          \\"deployment_ready_option\\": {
            \\"action_on_timeout\\": \\"CONTINUE_DEPLOYMENT\\"
          },
          \\"terminate_blue_instances_on_deployment_success\\": {
            \\"action\\": \\"TERMINATE\\",
            \\"termination_wait_time_in_minutes\\": 5
          }
        },
        \\"deployment_config_name\\": \\"CodeDeployDefault.ECSAllAtOnce\\",
        \\"deployment_group_name\\": \\"Test-Dev-ECS\\",
        \\"deployment_style\\": {
          \\"deployment_option\\": \\"WITH_TRAFFIC_CONTROL\\",
          \\"deployment_type\\": \\"BLUE_GREEN\\"
        },
        \\"ecs_service\\": {
          \\"cluster_name\\": \\"cluster\\",
          \\"service_name\\": \\"theService\\"
        },
        \\"load_balancer_info\\": {
          \\"target_group_pair_info\\": {
            \\"prod_traffic_route\\": {
              \\"listener_arns\\": [
                \\"listen-to-me\\"
              ]
            },
            \\"target_group\\": [
              {
                \\"name\\": \\"target-1\\"
              },
              {
                \\"name\\": \\"target-2\\"
              }
            ]
          }
        },
        \\"service_role_arn\\": \\"\${aws_iam_role.testCodeDeploy_ecs_code_deploy_role_EF679F94.arn}\\"
      }
    },
    \\"aws_codestarnotifications_notification_rule\\": {
      \\"testCodeDeploy_ecs_codedeploy_notifications_7223DA1D\\": {
        \\"detail_type\\": \\"BASIC\\",
        \\"event_type_ids\\": [
          \\"codedeploy-application-deployment-failed\\",
          \\"codedeploy-application-deployment-succeeded\\",
          \\"codedeploy-application-deployment-started\\"
        ],
        \\"name\\": \\"\${aws_codedeploy_app.testCodeDeploy_ecs_code_deploy_C4535318.name}\\",
        \\"resource\\": \\"arn:aws:codedeploy:\${data.aws_region.testCodeDeploy_current_region_B8533BBF.name}:\${data.aws_caller_identity.testCodeDeploy_current_account_0C0AC491.account_id}:application:\${aws_codedeploy_app.testCodeDeploy_ecs_code_deploy_C4535318.name}\\",
        \\"target\\": [
          {
            \\"address\\": \\"notify-me\\"
          }
        ]
      }
    },
    \\"aws_iam_role\\": {
      \\"testCodeDeploy_ecs_code_deploy_role_EF679F94\\": {
        \\"assume_role_policy\\": \\"\${data.aws_iam_policy_document.testCodeDeploy_codedeploy_assume_role_E9FEDA7F.json}\\",
        \\"name\\": \\"Test-Dev-ECSCodeDeployRole\\"
      }
    },
    \\"aws_iam_role_policy_attachment\\": {
      \\"testCodeDeploy_ecs_codedeploy_role_attachment_0C2B6B97\\": {
        \\"depends_on\\": [
          \\"aws_iam_role.testCodeDeploy_ecs_code_deploy_role_EF679F94\\"
        ],
        \\"policy_arn\\": \\"arn:aws:iam::aws:policy/AWSCodeDeployRoleForECS\\",
        \\"role\\": \\"\${aws_iam_role.testCodeDeploy_ecs_code_deploy_role_EF679F94.name}\\"
      }
    }
  }
}"
`;

exports[`ApplicationECSAlbCodeDeploy renders a CodeDeploy without a sns 1`] = `
"{
  \\"data\\": {
    \\"aws_iam_policy_document\\": {
      \\"testCodeDeploy_codedeploy_assume_role_E9FEDA7F\\": {
        \\"statement\\": [
          {
            \\"actions\\": [
              \\"sts:AssumeRole\\"
            ],
            \\"effect\\": \\"Allow\\",
            \\"principals\\": [
              {
                \\"identifiers\\": [
                  \\"codedeploy.amazonaws.com\\"
                ],
                \\"type\\": \\"Service\\"
              }
            ]
          }
        ]
      }
    }
  },
  \\"resource\\": {
    \\"aws_codedeploy_app\\": {
      \\"testCodeDeploy_ecs_code_deploy_C4535318\\": {
        \\"compute_platform\\": \\"ECS\\",
        \\"name\\": \\"Test-Dev-ECS\\"
      }
    },
    \\"aws_codedeploy_deployment_group\\": {
      \\"testCodeDeploy_ecs_codedeploy_deployment_group_F092EC5E\\": {
        \\"app_name\\": \\"\${aws_codedeploy_app.testCodeDeploy_ecs_code_deploy_C4535318.name}\\",
        \\"auto_rollback_configuration\\": {
          \\"enabled\\": true,
          \\"events\\": [
            \\"DEPLOYMENT_FAILURE\\"
          ]
        },
        \\"blue_green_deployment_config\\": {
          \\"deployment_ready_option\\": {
            \\"action_on_timeout\\": \\"CONTINUE_DEPLOYMENT\\"
          },
          \\"terminate_blue_instances_on_deployment_success\\": {
            \\"action\\": \\"TERMINATE\\",
            \\"termination_wait_time_in_minutes\\": 5
          }
        },
        \\"deployment_config_name\\": \\"CodeDeployDefault.ECSAllAtOnce\\",
        \\"deployment_group_name\\": \\"Test-Dev-ECS\\",
        \\"deployment_style\\": {
          \\"deployment_option\\": \\"WITH_TRAFFIC_CONTROL\\",
          \\"deployment_type\\": \\"BLUE_GREEN\\"
        },
        \\"ecs_service\\": {
          \\"cluster_name\\": \\"cluster\\",
          \\"service_name\\": \\"theService\\"
        },
        \\"load_balancer_info\\": {
          \\"target_group_pair_info\\": {
            \\"prod_traffic_route\\": {
              \\"listener_arns\\": [
                \\"listen-to-me\\"
              ]
            },
            \\"target_group\\": [
              {
                \\"name\\": \\"target-1\\"
              },
              {
                \\"name\\": \\"target-2\\"
              }
            ]
          }
        },
        \\"service_role_arn\\": \\"\${aws_iam_role.testCodeDeploy_ecs_code_deploy_role_EF679F94.arn}\\"
      }
    },
    \\"aws_iam_role\\": {
      \\"testCodeDeploy_ecs_code_deploy_role_EF679F94\\": {
        \\"assume_role_policy\\": \\"\${data.aws_iam_policy_document.testCodeDeploy_codedeploy_assume_role_E9FEDA7F.json}\\",
        \\"name\\": \\"Test-Dev-ECSCodeDeployRole\\"
      }
    },
    \\"aws_iam_role_policy_attachment\\": {
      \\"testCodeDeploy_ecs_codedeploy_role_attachment_0C2B6B97\\": {
        \\"depends_on\\": [
          \\"aws_iam_role.testCodeDeploy_ecs_code_deploy_role_EF679F94\\"
        ],
        \\"policy_arn\\": \\"arn:aws:iam::aws:policy/AWSCodeDeployRoleForECS\\",
        \\"role\\": \\"\${aws_iam_role.testCodeDeploy_ecs_code_deploy_role_EF679F94.name}\\"
      }
    }
  }
}"
`;
